util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/WindingOrder.java

```diff
@@ -76,7 +77,8 @@ else if (geometry instanceof Polygon) {
 		else if (geometry instanceof LinearRing) {
 			return unifyWindingOrderForLinearRing((LinearRing) geometry, counterClockWise);
 		}
-		else if (geometry instanceof GeometryCollection) {
+		else if (geometry instanceof GeometryCollection
+				&& geometry.getClass() == GeometryCollection.class) {
```

Comment:
If you do the class check you can omit the `instanceof` check. But I would prefer if you use equals instead of `==`, because in OSGi you might actually have the same class provided by different class loaders.
So something like this should work in all cases:
`GeometryCollection.class.equals(geometry.getClass())`